internal/models/interfaceconfig.go

```diff
@@ -0,0 +1,66 @@
+package models
+
+import (
+	"bytes"
+	"text/template"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+var (
+	tunnelTemplate = `[Interface]
+PrivateKey = {{.privateKey}}
+Address = {{.address}}
+
+{{range .interfacePeers}}
+[Peer]
+PublicKey = {{.PublicKey}}
+AllowedIPs = {{.AllowedIPs}}
+Endpoint = {{.Endpoint}}
+{{end}}`
+)
+
+// InterfaceConfig type represents a WireGuard tunnel configuration.
+type InterfaceConfig struct {
+	privateKey     string
+	address        string
+	interfacePeers []wgtypes.Peer
```

Comment:
This should be an internal type since want to add additional information to it (namely the friendly name/owner).  Though the embedded type can of course embed the wgtypes.Peer if you believe that is appropriate.  I'd rather it didn't though, since that will mean needing to write custom serializers for the embedded net.IP slices.

Reply:
Peers are already contained within the Device that is embedded within the drivers. Should we embed again here, or leave this type as a template handler and pull the information from there?

Reply:
I opted to use the Config type, since it holds multiple PeerConfigs. It seemed the best fit for accounting for multiple Peers. Let me know if you agree.

Reply:
The problem is more that this type contains types that can't be directly serialized by json, which makes it very difficult to write an API client around this.

What if you don't embed that in this struct at all, but instead have a method that lets you get a wgtypes type out.  So you'd have something like:

```go
type InterfaceConfig struct {
  Key     string
  Address string
  Peers   []InterfacePeer
  DNS     []string
  Domain  string
}

type InterfacePeer struct {
  Name       string
  Key        string
  AllowedIPs []string
  Endpoint   string
}
```

Note as well how these are all exported types, which is necessary for a marshaller to be able to marshal and unmarshal this type.  To get to the other types, you can implement methods on these types that return those.